diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0fe939d..f3ad0ed 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -229,7 +229,7 @@ endif()
 #============================================================================
 # Example binaries
 #============================================================================
-
+if(False) # lzs: disable-examples
 add_executable(example test/example.c)
 target_link_libraries(example zlib)
 add_test(example example)
@@ -247,3 +247,4 @@ if(HAVE_OFF64_T)
     target_link_libraries(minigzip64 zlib)
     set_target_properties(minigzip64 PROPERTIES COMPILE_FLAGS "-D_FILE_OFFSET_BITS=64")
 endif()
+endif()
